extensions/optaplanner-jsonb/runtime/src/main/java/io/quarkus/optaplanner/jsonb/OptaPlannerJsonbConfigCustomizer.java

```diff
@@ -0,0 +1,21 @@
+package io.quarkus.optaplanner.jsonb;
+
+import io.quarkus.jsonb.JsonbConfigCustomizer;
+import org.optaplanner.core.api.score.Score;
+import org.optaplanner.persistence.jsonb.api.OptaPlannerJsonbConfig;
+
+import javax.inject.Singleton;
+import javax.json.bind.JsonbConfig;
+
+/**
+ * OptaPlanner doesn't use JSON-B, but it does have optional JSON-B support for {@link Score}, etc.
+ */
+@Singleton
+public class OptaPlannerJsonbConfigCustomizer implements JsonbConfigCustomizer {
+
+    @Override
+    public void customize(JsonbConfig config) {
+        config = OptaPlannerJsonbConfig.createConfig();
```

Comment:
This is wierd. Does it work?

It doesn't modify the incoming config parameter, it reassigns it. Does this do anything?

Reply:
The integration test proves that this works. I have no idea how it can work though.

Reply:
@cuijulian Does the integration test fail if you uncomment this line?

Reply:
Indeed, this looks quite peculiar...

Reply:
@ge0ffrey The integration test still passes when I comment out that line. My thought process was to customize the `JsonbConfig` into the `OptaPlannerJsonbConfig` (which contains all the Score adapters) by reassigning it, but it seems to have no effect on the integration test. Alternatively, I could also manually customize the `config` using `withAdapters(insert all OptaPlanner Adapters here)`, but I doubt that would have an effect either. @geoand Any thoughts on why this could be?

Separate question: when I build the module with `-Pnative`, the `NativeImageTest` fails with:
```
[ERROR]   OptaPlannerTestResourceIT>OptaPlannerTestResourceTest.solveWithSolverFactory:26 1 expectation failed.
Response body doesn't match expectation.
Expected: is "{\"entityList\":[{\"value\":\"v1\"},{\"value\":\"v2\"}],\"score\":{\"initScore\":0,\"solutionInitialized\":true,\"score\":0},\"valueList\":[\"v1\",\"v2\"]}"
  Actual: {"entityList":[{"value":"v1"},{"value":"v2"}],"score":{"score":0},"valueList":["v1","v2"]}
```
How come the non native integration test passes, but the native IT gives a differently formatted response?

Reply:
@cuijulian I am not sure I understand you issue with the `JsonbConfigCustomizer`, can you elaborate please?

As for the test issue, I am not really sure. I would need to look into it, it could be a number of things.
Remind me to look at it tomorrow please as I will surely forget :)

Reply:
@geoand Basically customizing the `JsonbConfig` config parameter with `JsonbConfigCustomizer` has no effect on the `OptaPlannerTestResourceTest` integration test. 

As for the test issue, I'm going to do some more digging, I'll ping you if I have further questions. Thanks! :)

Reply:
@geoand @ge0ffrey Fixed by modifying the `config` using `withAdapters()`, the integration test works with expected output, for both non native and native mode.

Reply:
Excellent!